### DISCUSSION TITLE
How to know which Windows SDK APIs are being used?

### DISCUSSION THREAD

**Discussion by stevenbrix** [microsoft/win32metadata#1906] [Q&A]
Hey, I wasn't able to find an answer when looking through other discussions/docs, so I figured I'd ask here.

I'm wondering about the versioning scheme of the `Microsoft.Windows.SDK.Win32Metadata` package. The early versions matched the Windows SDK version that the metadata was for. This made it easy to understand which version of the SDK you were pulling in. But now, the packages follow a different versioning scheme which isn't as easy to follow.

How do I know which Windows SDK headers are being used?

@riverar @mikebattista I would appreciate some insight here!

**Comment by mikebattista** [comment]
You should assume the latest SDK with some lag between new SDK releases and metadata updates. If there's ever an API in a new SDK that you need that's missing from the metadata, please file a bug to encourage us to update. Usually, updating the SDK is pretty trivial, but we hit a snag with the most recent SDK (https://github.com/microsoft/win32metadata/issues/1930), so we are still behind on that one.

If you want to know specifically which SDK is used, you can see the dependency below. Switch to whatever release tag you're interested in to see what the dependency was at the time that metadata package was published.

https://github.com/microsoft/win32metadata/blob/c2b7ea95b5ae31b76db8f45bbe353b48a58a65f7/BuildTools/BuildTools.proj#L11-L12
